Core/Requirement/RequirementFactory.cs

```diff
@@ -100,6 +113,11 @@ public Requirement GetRequirement(string name, string requirement)
                 return CreateSpellInRangeRequirement(requirement);
             }
 
+            if (requirement.Contains("TargetCastingSpell:"))
```

Comment:
I would recommend to use `StartsWith` instead of `Contains`. And what you could do is to make a general

`{"TargetCastingSpell", () => playerReader.IsTargetCasting },`
in the 
https://github.com/Xian55/WowClassicGrindBot/pull/142/files#diff-0a5bea55e1ad484d71c7b843024683cb1b7d325e12fabaa928707f47f56a809eR123 

So the `TargetCastingSpell` still relevant just dosen't care about which spell is being casted. like as a paladin you can't interrupt the spell but if you have `Hammer of Justice` you can stun the target.